### PR TITLE
refactor!: Roll back to old `LSP1UniversalReceiverDelegate` Interface and functions

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?

## ♻️ Refactor

-  Add `UniversalReceiverDelegate` Interface and functions
- InterfaceId of LSP1Delegate: `0xa245bbda`

The UniversalReceiverDelegate will include the `universalReceiverDelegate` function having two additional parameter than `universalReceiver` which are the:
- caller: the address who called the `universalReceiver`
- value: the amount sent by the address who called the `universalReceiver`

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 1]
```diff
@@ -8,6 +8,10 @@ import {
     ILSP1UniversalReceiver
 } from "../LSP1UniversalReceiver/ILSP1UniversalReceiver.sol";
 
+import {
+    ILSP1UniversalReceiverDelegate
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:12]: Maybe we can alias if it shortens the line length and improve readability?

```suggestion
    ILSP1UniversalReceiverDelegate as ILSP1Delegate
```

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 2]
```diff
@@ -434,16 +439,16 @@ abstract contract LSP0ERC725AccountCore is
...
+                    _INTERFACEID_LSP1_DELEGATE
                 )
             ) {
-                // calling {universalReceiver} function on URD appending the caller and the value sent
-                resultDefaultDelegate = universalReceiverDelegate
-                    .callUniversalReceiverWithCallerInfos(
-                        typeId,
-                        receivedData,
+                resultDefaultDelegate = ILSP1UniversalReceiverDelegate(
+                    universalReceiverDelegate
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:446]: Would that make it more readable and shorter?

```solidity
resultDefaultDelegate = ILSP1Delegate(lsp1Delegate).universalReceiverDelegate(
    msg.sender,
    msg.value,
    typeId,
    receivedData
);
```

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol [hunk 3]
```diff
@@ -76,17 +78,12 @@ contract LSP1UniversalReceiverDelegateUP is ERC165, ILSP1UniversalReceiver {
...
         bytes memory /* data */
-    ) public payable virtual override returns (bytes memory) {
-        // CHECK that we did not send any native tokens to the LSP1 Delegate, as it cannot transfer them back.
-        if (msg.value != 0) {
-            revert NativeTokensNotAccepted();
-        }
-
-        address notifier = address(bytes20(msg.data[msg.data.length - 52:]));
-
+    ) public virtual returns (bytes memory) {
```
Comment (CJ42) [inline contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:86]: Do we want to keep this function not payable? Would there be a use case to make it payable?

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol [hunk 4]
```diff
@@ -53,7 +52,10 @@ import {
  * - Writes the data keys representing the owned vaults from type [LSP-9-Vault] into your account storage, and removes them when transferring ownership to other accounts according to the [LSP-10-ReceivedVaults] Standard.
  *
  */
-contract LSP1UniversalReceiverDelegateUP is ERC165, ILSP1UniversalReceiver {
+contract LSP1UniversalReceiverDelegateUP is
+    ERC165,
+    ILSP1UniversalReceiverDelegate
```
Comment (CJ42) [inline contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:57]: interface should go before `ERC165` in the inheritance. From most base (left) to most derived (right)

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol (hunk 3 above)
Comment (CJ42) [reply contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:86]: It is also missing the `override` keyword, so that it can compile with `solc < 0.8.8`

File: contracts/Mocks/UniversalReceivers/UniversalReceiverDelegateDataUpdater.sol [hunk 5]
```diff
@@ -15,14 +15,18 @@ import {
     _TYPEID_LSP7_TOKENSSENDER
 } from "../../LSP7DigitalAsset/LSP7Constants.sol";
 
-import {_INTERFACEID_LSP1} from "../../LSP1UniversalReceiver/LSP1Constants.sol";
+import {
+    _INTERFACEID_LSP1_DELEGATE
+} from "../../LSP1UniversalReceiver/LSP1Constants.sol";
 
 contract UniversalReceiverDelegateDataUpdater is ERC165Storage {
```
Comment (CJ42) [inline contracts/Mocks/UniversalReceivers/UniversalReceiverDelegateDataUpdater.sol:22]: We could add the `ILSP1Delegate` in the inheritance here, even if it is a Mock contract.

File: contracts/Mocks/UniversalReceivers/UniversalReceiverDelegateGasConsumer.sol [hunk 6]
```diff
@@ -3,27 +3,31 @@ pragma solidity ^0.8.4;
...
      */
-    function universalReceiver(
-        bytes32 /* typeId */,
+    function universalReceiverDelegate(
+        address /*sender*/,
+        uint256 /*value*/,
+        bytes32 /*typeId*/,
         bytes memory /* data */
-    ) public payable override returns (bytes memory) {
+    ) public virtual returns (bytes memory) {
```
Comment (CJ42) [inline contracts/Mocks/UniversalReceivers/UniversalReceiverDelegateGasConsumer.sol:30]: same comment about old solc compiler version (otherwise CI will fail once it is merge into `develop`).

File: tests/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP.behaviour.ts [hunk 7]
```diff
@@ -3167,22 +3167,4 @@ export const shouldInitializeLikeLSP1Delegate = (
       );
     });
   });
-  describe('edge cases', () => {
```
Comment (CJ42) [inline tests/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP.behaviour.ts:3170]: We could keep this test to ensure that we cannot send value to the `universalReceiverDelegate` function maybe?

File: scripts/interfaceIds.ts [hunk 8]
```diff
@@ -33,14 +33,19 @@ async function main() {
 
       // adjust the source path for LSP20 and LSP17 contracts
```
Comment (CJ42) [inline scripts/interfaceIds.ts:34]:
```suggestion
      // adjust the source path for LSP1Delegate, LSP20 and LSP17 contracts
```

**Review by CJ42 — CHANGES_REQUESTED** [review]
Let's make sure the `universalReceiverDelegate` functions (in both the contracts/ and the Mock contracts) that inherit the ILSP1Delegate interface have the `override` keyword, so that they can be compiled with old solc versions and so that the CI passes once it is merged ✅

File: tests/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP.behaviour.ts (hunk 7 above)
Comment (YamenMerhi) [reply tests/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP.behaviour.ts:3170]: you don't have the option to send value when the function is not payable

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol (hunk 4 above)
Comment (YamenMerhi) [reply contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:57]: @CJ42 I think most base is `ERC165` and most derived is `ILSP1`

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol (hunk 3 above)
Comment (YamenMerhi) [reply contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:86]: I don't think it should be payable

**Review by CJ42 — APPROVED** [review]

File: contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol (hunk 4 above)
Comment (CJ42) [reply contracts/LSP1UniversalReceiver/LSP1UniversalReceiverDelegateUP/LSP1UniversalReceiverDelegateUP.sol:57]: We can leave it like that for now until we add tests to check the inheritance graph + linearization.

**Review by b00ste — APPROVED** [review]

**Review by skimaharvey — APPROVED** [review]
